From cedd018f2cf5ce6556014d9d7092f29180de69c7 Mon Sep 17 00:00:00 2001
From: Koushik Chakravarty <koushik.chakravarty@citrix.com>
Date: Tue, 22 Aug 2023 10:01:27 +0000
Subject: [PATCH] CP-9895-Add-originator-to-login_with_password-xapi-call

If the XAPI client passes the originator parameter to the
login_with_password API, it gives the client its own pool
of XAPI sessions, thereby protecting its sessions from
other clients who misbehave or use too many sesions.
Also it helps debugging and isolating xapi sessions.
---
 tests/MockXenAPI.py                                     | 2 +-
 xenserver/opt_xensource_libexec_InterfaceReconfigure.py | 2 +-
 xenserver/usr_share_openvswitch_scripts_ovs-xapi-sync   | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/MockXenAPI.py b/tests/MockXenAPI.py
index b8de4ee71..444569d1c 100644
--- a/tests/MockXenAPI.py
+++ b/tests/MockXenAPI.py
@@ -35,7 +35,7 @@ class XenAPI(object):
         self.VIF = VIF()
         self.VM = VM()
 
-    def login_with_password(self, unused_username, unused_password):
+    def login_with_password(self, unused_username, unused_password, unused_version, unused_originator):
         pass
 
 
diff --git a/xenserver/opt_xensource_libexec_InterfaceReconfigure.py b/xenserver/opt_xensource_libexec_InterfaceReconfigure.py
index 74b784d34..bbe510259 100644
--- a/xenserver/opt_xensource_libexec_InterfaceReconfigure.py
+++ b/xenserver/opt_xensource_libexec_InterfaceReconfigure.py
@@ -495,7 +495,7 @@ class DatabaseCache(object):
 
             if not session_ref:
                 log("No session ref given on command line, logging in.")
-                session.xenapi.login_with_password("root", "")
+                session.xenapi.login_with_password("root", "", "", "ovs-interface-reconfigure")
             else:
                 session._session = session_ref
 
diff --git a/xenserver/usr_share_openvswitch_scripts_ovs-xapi-sync b/xenserver/usr_share_openvswitch_scripts_ovs-xapi-sync
index bff85464b..ce6fef493 100755
--- a/xenserver/usr_share_openvswitch_scripts_ovs-xapi-sync
+++ b/xenserver/usr_share_openvswitch_scripts_ovs-xapi-sync
@@ -66,7 +66,7 @@ def init_session():
 
     try:
         session = XenAPI.xapi_local()
-        session.xenapi.login_with_password("", "")
+        session.xenapi.login_with_password("", "", "", "ovs-xapi-sync")
     except XenAPI.Failure as e:
         session = None
         vlog.warn("Couldn't login to XAPI (%s)" % e)
-- 
2.34.1

